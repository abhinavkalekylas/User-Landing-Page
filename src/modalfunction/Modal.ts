export interface User {
  id: number;
  name: string;
  email: string;
  gender: GenderType;
  status: StatusType;
}

export enum GenderType {
  MALE = "male",
  FEMALE = "female",
}

export enum StatusType {
  ACTIVE = "active",
  INACTIVE = "inactive",
}

export const validateEmail = (email: string) => {
  const regex = /^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/i;
  if (regex.test(email)) {
    return true;
  }
  return false;
};
