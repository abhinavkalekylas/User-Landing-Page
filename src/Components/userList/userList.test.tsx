import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { token } from "../../modalfunction/Modal";
import UserList from "./userList";
import axios from "axios";

jest.mock("axios");

const data = [
  {
    id: 3218,
    name: "Harinarayan Abbott",
    email: "abbott_harinarayan@collins.info",
    gender: "male",
    status: "inactive",
  },
];

var MockAdapter = require("axios-mock-adapter");
var mock = new MockAdapter(axios);

let wrapper: any;
beforeEach(() => {
  wrapper = shallow(<UserList />);
});

afterEach(() => {
  jest.clearAllMocks();
});

// snapshot testing
it("renders correctly UserList component", () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("test axios", async () => {
  // Mock any GET request
  // arguments for reply are (status, data, headers)
  mock.onGet("https://gorest.co.in/public/v2/users/").reply(200, data, {
    "Content-Type": "application/json",
    Authorization: `Bearer ${token}`,
  });

  const res = await axios.get("https://gorest.co.in/public/v2/users/", {
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${token}`,
    },
  });

  expect(res.data).toEqual(data);
});

// it("test axios", async () => {
//   var MockAdapter = require("axios-mock-adapter");

//   // This sets the mock adapter on the default instance
//   var mock = new MockAdapter(axios);

//   const data = [{
//       "id": 3218,
//       "name": "Harinarayan Abbott",
//       "email": "abbott_harinarayan@collins.info",
//       "gender": "male",
//       "status": "inactive"
//   }]

//   // Mock any GET request
//   // arguments for reply are (status, data, headers)
//   mock.onGet("https://gorest.co.in/public/v2/users/").reply(200,
//     data, {
//       "Content-Type": "application/json",
//       Authorization: `Bearer ${token}`,
//   });

//   let wrapper:any;
//   act(() => {
//     wrapper = shallow(<UserList />)
//   })

//   wrapper.update();
//   expect(toJson(wrapper)).toMatchSnapshot()
// })

it("On click edit user test", () => {
  //call props boolean
  wrapper.find(".edit_btn").simulate("click");
  expect(wrapper.find("EditUser").props().editUserModal).toBe(true);
});

it("On click view user test", () => {
  //call props boolean
  wrapper.find(".view_btn").simulate("click");
  expect(wrapper.find("ViewUser").props().viewUserModal).toBe(true);
});
