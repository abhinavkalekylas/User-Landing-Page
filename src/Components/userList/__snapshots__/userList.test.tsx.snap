// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly UserList component 1`] = `
<div
  className="userlist_container"
>
  <table
    className="table table-light"
  >
    <thead>
      <tr>
        <th
          className="title_name"
          scope="col"
        >
          Name
        </th>
        <th
          scope="col"
        >
          Email
        </th>
        <th
          className="action"
          scope="col"
        >
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="table_row"
        key="0"
      >
        <td
          className="user_name"
        />
        <td
          className="user_email"
        />
        <td
          className="action_buttons"
        >
          <Button
            className="view_btn view-button text-white"
            color="info"
            onClick={[Function]}
            tag="button"
          >
            View User
          </Button>
          <Viewuser
            closeModal={[Function]}
            userData={
              Object {
                "email": "",
                "gender": "male",
                "id": 0,
                "name": "",
                "status": "active",
              }
            }
            viewUserModal={false}
          />
          <Button
            className="edit_btn edit-button"
            color="primary"
            onClick={[Function]}
            tag="button"
          >
            Edit User
          </Button>
          <EditUser
            closeModal={[Function]}
            editUserModal={false}
            userData={
              Object {
                "email": "",
                "gender": "male",
                "id": 0,
                "name": "",
                "status": "active",
              }
            }
          />
          <Button
            className="delete-button"
            color="danger"
            tag="button"
          >
            Delete User
          </Button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
